neuro_flow/batch_runner.py

```diff
@@ -490,9 +484,20 @@ async def _restart(
                     graph = graphs[prefix]
                     deps = graph[task.id]
                     if not deps or all(dep in handled for dep in deps):
+                        if (
+                            prefix in finished
+                            and finished[prefix].status != TaskStatus.SUCCEEDED
```

Comment:
How FAILED tasks are handled?

Reply:
I did not get your question, sorry.
I rechecked this place and looks like there is a bug in master. For example, if there is an action with two tasks, and the first one succeeded (and it has no deps), and the second fails, then this code will copy the first task of action, but it will not copy `StartedTask` for the action itself. And it is a problem because such data is inconsistent (action task cannot be started before the action started).

Reply:
Nevermind, thanks